pub/js/main.js

```diff
@@ -179,25 +195,38 @@ $(function () {
 					if (response.error_code === 0){
 						accordion.setCurrent('#step-finalize');
 					}
-					handleResponse(response, function () {}, '#step-finalize');
+					handleResponse(response, function () {}, '#step-appupgrade');
 					return response.error_code === 0
 							? $.post($('#meta-information').data('endpoint'), {command: 'upgrade:restartWebServer'})
 							: $.Deferred()
 							;
 				})
 				.then(function (response) {
-					handleResponse(response, function () {}, '#step-finalize');
-					return response.error_code === 0
-							? $.post($('#meta-information').data('endpoint'), {command: 'upgrade:postUpgradeCleanup'})
+					handleResponse(
+						response, 
+						function () {
+							accordion.showContent('#step-finalize');
+						},
+						'#step-finalize'
+					);
+					return (response.error_code === 0
+							?  accordion.setCurrent('#step-finalize')
+							  || $.post($('#meta-information').data('endpoint'), {command: 'upgrade:postUpgradeCleanup'})
 							: $.Deferred()
-							;
+					);
 				})
 				.then(function (response) {
 					handleResponse(response, function () {}, '#step-finalize');
 					if (response.error_code === 0){
 						accordion.setDone('#step-done');
-						accordion.setContent('#step-done', 'All done!');
+						accordion.setContent('#step-done', 'All done! Redirecting you to your ownCloud.');
```

Comment:
use "product name" here if possible due to theming ? probably irrelevant for the updater anyway...